map/gps_track.cpp

```diff
@@ -351,3 +359,27 @@ void GpsTrack::NotifyCallback(pair<size_t, size_t> const & addedIds, pair<size_t
     m_callback(std::move(toAdd), evictedIds);
   }
 }
+
+std::vector<location::GpsTrackInfo> GpsTrack::GetCurrentTrack() const
```

Comment:
nit: A lambda can be passed here and run in ForEach below, or something like this (in .hpp):

```suggestion
template <typename F>
void ForEachPoint(F && f) const
{
  m_collection->ForEach(std::move(f));
}
```